File: sdk/javascript/examples/register-live.mjs
/**
 * Live registration test — signs a receipt with the SDK and posts it
 * to the real production anchor service.
 *
 * Run: node examples/register-live.mjs
 */
import { seal, verify, register, generateKeySync } from "../dist/index.js";

const key = generateKeySync();
console.log("[test] generated key, signer =", key.publicHex.slice(0, 20) + "…");

const payload = {
  test: "live-register",
  timestamp_local: new Date().toISOString(),
  random: Math.floor(Math.random() * 1e9),
};

const r = await seal(payload, { key, payloadType: "test/live" });
console.log("[test] sealed:", r.id);

const v = await verify(r, payload);
console.log("[test] local verify:", v.valid ? "VALID" : "INVALID");
if (!v.valid) {
  console.error("[test] cannot register an invalid local seal");
  process.exit(1);
}

console.log("[test] posting to https://croviatrust.com/api/anchor …");
const result = await register(r);
console.log("[test] register result:", result);

if (!result.accepted) {
  console.error("[test] registration FAILED");
  process.exit(1);
}

// Idempotency: posting the same receipt again must succeed and return the same anchor.
const result2 = await register(r);
console.log("[test] re-register (idempotent):", result2);

// Tampered receipt must be rejected.
const tampered = { ...r, sig: r.sig.replace(/^./, (c) => (c === "0" ? "1" : "0")) };
const result3 = await register(tampered);
console.log(
  "[test] register tampered (must reject):",
  result3.accepted ? "ACCEPTED ✗ BUG" : `REJECTED ✓ (${result3.error})`,
);

File: receipt/python/examples/register_live.py
"""
Live registration test — signs a receipt with the Python SDK and posts
it to the real production anchor service.

Run:  python examples/register_live.py
"""
import sys
import time

from crovia_seal import generate_key, register, seal, verify


def main() -> int:
    key = generate_key()
    print(f"[test] generated key, signer = {key.public_hex[:20]}…")

    payload = {
        "test": "live-register-py",
        "timestamp_local": time.strftime("%Y-%m-%dT%H:%M:%S.000Z"),
        "random": int.from_bytes(__import__("os").urandom(4), "big"),
    }

    r = seal(payload, key=key, payload_type="test/live-py")
    print(f"[test] sealed: {r['id']}")

    v = verify(r, payload)
    print(f"[test] local verify: {'VALID' if v.valid else 'INVALID'}")
    if not v.valid:
        print("[test] cannot register an invalid local seal")
        return 1

    print("[test] posting to https://croviatrust.com/api/anchor …")
    ack = register(r)
    print(f"[test] register result: accepted={ack.accepted} status={ack.status} "
          f"anchor_id={ack.anchor_id} duplicate={ack.duplicate}")

    if not ack.accepted:
        print(f"[test] registration FAILED: {ack.error}")
        return 1

    # Idempotency.
    ack2 = register(r)
    print(f"[test] re-register (idempotent): accepted={ack2.accepted} "
          f"anchor_id={ack2.anchor_id} duplicate={ack2.duplicate}")

    # Tampered receipt must be rejected.
    tampered = dict(r)
    sig = r["sig"]
    tampered["sig"] = ("1" if sig[0] == "0" else "0") + sig[1:]
    ack3 = register(tampered)
    if ack3.accepted:
        print("[test] register tampered: ACCEPTED ✗ BUG")
        return 1
    print(f"[test] register tampered (must reject): REJECTED ✓ ({ack3.error})")
    return 0


if __name__ == "__main__":
    sys.exit(main())
